server/gitlab/api.go

```diff
@@ -314,15 +351,113 @@ func (g *gitlab) GetYourPrs(ctx context.Context, user *UserInfo) ([]*internGitla
 		return nil, err
 	}
 
-	return result, nil
+	var mergeRequests []*MergeRequest
+	for _, res := range result {
+		if res.Labels != nil {
+			var labelsWithDetails []*internGitlab.Label
+			labelsWithDetails, err = g.GetLabelDetails(client, res.ProjectID, res.Labels)
+			if err != nil {
+				return nil, err
+			}
+			mergeRequest := &MergeRequest{
+				MergeRequest:      res,
+				LabelsWithDetails: labelsWithDetails,
+			}
+			mergeRequests = append(mergeRequests, mergeRequest)
+		}
+	}
+	return mergeRequests, nil
+}
+
+func (g *gitlab) GetLabelDetails(client *internGitlab.Client, pid int, labels internGitlab.Labels) ([]*internGitlab.Label, error) {
+	var labelsWithDetails []*internGitlab.Label
+	for _, label := range labels {
+		labelWithDetails, resp, err := client.Labels.GetLabel(pid, label)
+		if respErr := checkResponse(resp); respErr != nil {
+			return nil, respErr
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "can't get label in GitLab api")
+		}
+		labelsWithDetails = append(labelsWithDetails, labelWithDetails)
+	}
+
+	return labelsWithDetails, nil
 }
 
-func (g *gitlab) GetYourAssignments(ctx context.Context, user *UserInfo) ([]*internGitlab.Issue, error) {
+func (g *gitlab) GetYourPrDetails(ctx context.Context, log logger.Logger, user *UserInfo, prList []*PRDetails) ([]*PRDetails, error) {
 	client, err := g.gitlabConnect(*user.Token)
 	if err != nil {
 		return nil, err
 	}
 
+	var result []*PRDetails
+	var wg sync.WaitGroup
+	for _, pr := range prList {
+		wg.Add(1)
+		go func(pid, iid int, sha string) {
+			defer wg.Done()
+			res := g.fetchYourPrDetails(ctx, log, client, pid, iid, sha)
+			if res != nil {
+				result = append(result, res)
+			}
+		}(pr.ProjectID, pr.IID, pr.SHA)
+	}
+	wg.Wait()
+	return result, nil
+}
+
+func (g *gitlab) fetchYourPrDetails(c context.Context, log logger.Logger, client *internGitlab.Client, pid, iid int, sha string) *PRDetails {
```

Comment:
Yes, technically you are correct. But the previous implementation consisted of `PR` (like function `getYourPrs`), so to keep in sync with them we also kept the name `PR` instead of `MR`.